src/group_gui.cpp

```diff
@@ -851,11 +875,32 @@ class VehicleGroupWindow : public BaseVehicleListWindow {
 				this->SetDirty();
 
 				uint id_v = this->vscroll->GetScrolledRowFromWidget(pt.y, this, WID_GL_LIST_VEHICLE);
-				if (id_v >= this->vehicles.size()) return; // click out of list bound
+				if (id_v >= this->vehgroups.size()) return; // click out of list bound
+
+				const GUIVehicleGroup &vehgroup = this->vehgroups[id_v];
+				switch (this->grouping) {
+					case GB_NONE: {
+						const Vehicle *v = vehgroup.GetSingleVehicle();
+						if (!VehicleClicked(v) && vindex == v->index) {
+							ShowVehicleViewWindow(v);
+						}
+						break;
+					}
+
+					case GB_SHARED_ORDERS: {
+						const Vehicle *v = vehgroup.vehicles_begin[0];
+						/*
+						No VehicleClicked(v) support for now, because don't want
+						to enable any contextual actions except perhaps clicking/ctrl-clicking to clone orders.
+						*/
```

Comment:
Later in this patch this becomes a `TODO`. I explained a bit there why I find these comments odd .. "for now" , suggesting we can expect another PR soon? :)

(also another instance of this slightly above)

Reply:
Done